src/index.js

```diff
@@ -12,35 +12,27 @@ export { Bibliography };
 
 const __dirname = fileURLToPath(new URL("..", import.meta.url));
 
-function defaultRenderCitation (citationTemplate) {
-	if (citationTemplate) {
-		return info => nunjucks.render(citationTemplate, info);
-	}
-	else {
-		let template = fs.readFileSync(__dirname + "/_includes/_citations.njk", "utf8");
-		return info => nunjucks.renderString(template, info);
-	}
-}
-
 let doiTemplates = {
 	url: '<a href="$&" class="doi">$&</a>',
 	id: '<a href="https://doi.org/$1" class="doi">$1</a>',
 };
 
 export default function (config, {
-	citationTemplate,
-	citationRender = defaultRenderCitation(citationTemplate),
+	citationTemplate = __dirname + "/_includes/_citations.njk",
+	citationRender,
 	style, locale, // defaults set in Bibliography
 	bibliography: globalBibliography,
 } = {}) {
 	const references = new Bibliographies({globalBibliography, style, locale});
+	let fileRenderer;
+	const render = citationRender ?? (async function(...args){
+		fileRenderer ??= await RenderPlugin.File(citationTemplate);
+		return fileRenderer.call(this, ...args);
+	});
 
-	function renderCitations (content) {
+	async function renderCitations (content) {
 		let refs = references.getOrCreate(this.page, this.ctx.bibliography);
-
-		return citations.render(content, refs, {
-			render: citationRender,
-		});
+		return await citations.render(content, refs, { render });
```

Comment:
Isn’t `return await` practically the same as `return`? Or does 11ty do some clever stuff if the function passed to a filter is async?

Reply:
It functions basically the same, but Eleventy wants the `async` keyword for asynchronous filters (see [this comment](https://github.com/11ty/eleventy/blob/ff6338f2d2e7a9fce3ee5cb605c180758892bb13/src/UserConfig.js#L329) in `.addFilter()`).

I also usually do this for readability so I know that `renderCitations()` and `citations.render(…)` are async and return promises, but in practice it's functionally near-equivalent (differing only slightly with regards to errors). In other words, it's mostly a preference thing, I can remove the `await` here if you'd like.